test/views/behaviours/ViewBehaviours.scala

```diff
@@ -23,60 +23,82 @@ trait ViewBehaviours extends ViewSpecBase {
 
   val expectTimeoutDialog: Boolean = true
 
-  protected def normalPage(
-    view: HtmlFormat.Appendable,
-    messageKeyPrefix: String,
-    messageKeySuffix: String,
-    messageHeadingArgs: Any*
-  )(
+  val viewViaApply: HtmlFormat.Appendable
+  val viewViaRender: HtmlFormat.Appendable
+  val viewViaF: HtmlFormat.Appendable
+
+  def normalPage(messageKeyPrefix: String, messageKeySuffix: String, messageHeadingArgs: Any*)(
     expectedGuidanceKeys: String*
   ): Unit =
     "behave like a normal page" - {
-      val suffix = if (messageKeySuffix.isEmpty) "" else s".$messageKeySuffix"
+      Seq(".apply", ".render", ".f").foreach {
+        case byMethod @ ".apply"  =>
+          pageByMethod(viewViaApply, messageKeyPrefix, messageKeySuffix, messageHeadingArgs: _*)(byMethod)
+        case byMethod @ ".render" =>
+          pageByMethod(viewViaRender, messageKeyPrefix, messageKeySuffix, messageHeadingArgs: _*)(byMethod)
+        case byMethod @ ".f"      =>
+          pageByMethod(viewViaF, messageKeyPrefix, messageKeySuffix, messageHeadingArgs: _*)(byMethod)
+      }
```

Comment:
Can this be squashed further or something along those lines, thanks. Like the below:

```
      Seq((".apply", viewViaApply), (".render", viewViaRender), (".f", viewViaF)).foreach { case (method, view) =>
        pageByMethod(view, messageKeyPrefix, messageKeySuffix, messageHeadingArgs: _*)(method)
```